pandas/tests/groupby/test_groupby.py

```diff
@@ -806,7 +806,7 @@ def test_groupby_as_index_cython(df):
     msg = "The default value of numeric_only"
     with tm.assert_produces_warning(FutureWarning, match=msg):
         result = grouped.mean()
-        expected = data.groupby(["A"]).mean()
+        expected = data.groupby("A").mean()
```

Comment:
Why is this changing?

Reply:
As I mentioned in #47761 
I thought it's better to generalize the rule of using not using a list when grouping by a single key as groupby is being iterated over in other functions. So I thought it will be a good idea to generalize the rule.

Reply:
> I thought it's better to generalize the rule of...

I do not understand what this means. Can you expand on it? Also, it's not clear to me - is the result of `data.groupby(["A"]).mean()` different from what the main branch currently produces?